tensorflow/core/common_runtime/local_device.cc

```diff
@@ -114,9 +114,6 @@ struct LocalDevice::EigenThreadPoolInfo {
 LocalDevice::LocalDevice(const SessionOptions& options,
                          const DeviceAttributes& attributes)
     : Device(options.env, attributes), owned_tp_info_(nullptr) {
-  // Log info messages if TensorFlow is not compiled with instructions that
-  // could speed up performance and are available on the current CPU.
-  port::InfoAboutUnusedCPUFeatures();
```

Comment:
@cantonios Thank you for the quick reply!

I just realized @itaiz-google is OOO until after the branch cut. I wanted to check with him before adding `cpu_feature_guard.so` back, because he had originally removed it for ODR violation as well.

How about this? 
- We remove `_pywrap_cpu_feature_guard.InfoAboutUnusedCPUFeatures()` call from self_check.py, and 
- Guard the whole [InfoAboutUnusedCPUFeatures](https://github.com/tensorflow/tensorflow/blob/374730d691002609032eb6a1e86a78d3c529113d/tensorflow/core/platform/cpu_feature_guard.cc#L126-L190) function with an `#ifndef TF_PYTHON_CPU_FEATURE_GUARD`, then pass `-DTF_PYTHON_CPU_FEATURE_GUARD` as copt to the [_pywrap_cpu_feature_guard](https://github.com/tensorflow/tensorflow/blob/39964b4b80c172eec9f06f94c414f5d38b33bd67/tensorflow/python/platform/BUILD#L200) target

This way, the `_pywrap_cpu_feature_guard` target will only get code wrapped in an unnamed namespace and won't clash with the original C++ target. TF will also print only once.

Reply:
Hmm, that seems complicated.

Is the issue that the
```
absl::once_flag g_cpu_feature_guard_warn_once_flag;
```
flag is in an anonymous namespace?

Reply:
Since `cpu_feature_guard.cc` appeared in two seperate build targets ([cpu_feature_guard](https://github.com/tensorflow/tensorflow/blob/f2d2431210c007948767a5f2f5f2ae58d4de32e6/tensorflow/core/platform/BUILD#L199) and [_pywrap_cpu_feature_guard](https://github.com/tensorflow/tensorflow/blob/39964b4b80c172eec9f06f94c414f5d38b33bd67/tensorflow/python/platform/BUILD#L200)), there could be two [InfoAboutUnusedCPUFeatures](https://github.com/tensorflow/tensorflow/blob/374730d691002609032eb6a1e86a78d3c529113d/tensorflow/core/platform/cpu_feature_guard.cc#L126-L190) from two binaries. (The rest of the file is fine since it's in an anonymous namespace.) 

I would like to guard [InfoAboutUnusedCPUFeatures](https://github.com/tensorflow/tensorflow/blob/374730d691002609032eb6a1e86a78d3c529113d/tensorflow/core/platform/cpu_feature_guard.cc#L126-L190) with an ifdef so  [_pywrap_cpu_feature_guard](https://github.com/tensorflow/tensorflow/blob/39964b4b80c172eec9f06f94c414f5d38b33bd67/tensorflow/python/platform/BUILD#L200) doesn't have this function. It doesn't need it. It only needs the class `CPUFeatureGuard` to detect if the binary is compiled with an unsupported instruction (and die properly if so).

Reply:
@penpornk thanks for the explanation. Removing the line https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/platform/self_check.py#L64 seems to have the same effect - pywrap_cpu_feature_guard fails for unsupported instruction, but doesn't print the unused instructions.
Unused instruction is only reported from C++.  Would this solution be ok?

Reply:
Thank you for verifying, @agramesh1! Let's remove the `Info...` call in Python for now.
https://github.com/tensorflow/tensorflow/blob/f77466618610426c17dafbe243637efe6e7bb6de/tensorflow/python/platform/self_check.py#L64

This won't solve the fact that there will be two `Info...` functions in the overall lib. But it should be fine since we have been having this in TF 2.10 and TF 2.11 too. 

@cantonios and I were talking about refactoring the `CPUFeatureGuard` class to a header file and make the `_pywrap_cpu_feature_guard` target only depend on the header file. But since the branch cut is near and refactoring could potentially cause unexpected issues blocking the PR merge, I think I'll just do this later in a separate commit (likely after the branch cut).

Reply:
The issue is that each pywrap module is built into its own dynamic library.  So the `_pywrap_cpu_feature_guard` has one copy of the C++ `cpu_feature_guard` symbols, and the main tensorflow library has its own separate copy.  When calling the pywrap module, `InfoAboutUusedCPUFeatures()` uses the one in its own dynamic library, but when called from anywhere else, it uses the one in the main library.  Each has their own separate versions of variables to detect whether the function has previously been called, so we end up getting two different calls each thinking it is the first time, and the message is printed twice.

To avoid this, we can't have any source included in the main TF library to also be separately included in a pywrap module.  Instead the pywrap module can only depend on TF headers (excluding sources, no inline functions), and must be defined via `tf_python_pybind_extension`.  This will link in the main TF library so that it has all the right symbols.